lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -1679,10 +1730,229 @@ def test_import_val_data_internal(self):
         self.assertEqual(video_data['encoded_videos'][0]['url'], 'http://example.com/video')
         self.assertEqual(video_data['encoded_videos'][0]['file_size'], 222)
         self.assertEqual(video_data['encoded_videos'][0]['bitrate'], 333)
-        # verify transcript data
-        self.assertDictEqual(
-            get_video_transcript(video.edx_video_id, 'ar'),
-            self.get_video_transcript_data('test_edx_video_id')
+
+        # Verify that VAL transcript is imported.
+        self.assertDictContainsSubset(
+            self.get_video_transcript_data(
+                edx_video_id,
+                language_code=val_transcript_language_code,
+                provider=val_transcript_provider
+            ),
+            get_video_transcript(video.edx_video_id, val_transcript_language_code)
+        )
+
+        # Verify that transcript from sub field is imported.
+        self.assertDictContainsSubset(
+            self.get_video_transcript_data(
+                edx_video_id,
+                language_code=self.descriptor.transcript_language
+            ),
+            get_video_transcript(video.edx_video_id, self.descriptor.transcript_language)
+        )
+
+        # Verify that transcript from transcript field is imported.
+        self.assertDictContainsSubset(
+            self.get_video_transcript_data(
+                edx_video_id,
+                language_code=external_transcript_language_code
+            ),
+            get_video_transcript(video.edx_video_id, external_transcript_language_code)
+        )
+
+    def test_import_no_video_id(self):
+        """
+        Test that importing a video with no video id, creates a new external video.
+        """
+        xml_data = """<video><video_asset></video_asset></video>"""
+        module_system = DummySystem(load_error_modules=True)
+        id_generator = Mock()
+
+        video = self.descriptor.from_xml(xml_data, module_system, id_generator)
+        self.assertIsNotNone(video.edx_video_id)
```

Comment:
`edx_video_id` is str type and its default value is empty string. so for empty string `assertIsNotNone` will pass but it shouldn't.

we should also check `edx_video_id` before `self.descriptor.from_xml` method call above.